base/asyncevent.jl

```diff
@@ -284,7 +285,8 @@ function timedwait(testcb::Function, secs::Float64; pollint::Float64=0.1)
 
     if !testcb()
         t = Timer(timercb, pollint, interval = pollint)
-        ret = fetch(done)::Symbol
+        ret = fetch(done)
+        ret isa Exception && rethrow(ret)
```

Comment:
I don't think we should rely on the type of `e`; can use a boolean flag instead.

Reply:
How about if we store a `Tuple{Symbol, Union{Nothing, Exception}}` in the channel? Then this code would look like:
```julia
ret, e = fetch(done)
ret === :error && rethrow(e)
```